### PR TITLE
Integrate functional editor pane with schema/endpoint fetching. Refactor state and prop drilling. Reposition UI elements. Remove unnecessary code fragments.

### PR DESCRIPTION
**Pull request by clark-pang**
Removed unused or unnecessary bits of code. 
Refactored the structure of App and Editor to dovetail functionality.
Rearrange state and props to appropriately handle the logical flow from top down.
Rewrite parts of Editor to reflect these changes and handle oddities that resulted.
Debugged, and repositioned UI for bare-bones usage.

### CODE REVIEW

File: client/src/Editor.jsx [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+// add editor to DOM
+// ref is the element the editor will hook onto
+const createEditor = (
+  ref,
+  options,
+) => editor.create(ref.current, options);
+
+
+export default function Editor({schema, endpoint}) {
```
Comment (gacetta) [inline client/src/Editor.jsx:67]: The only thing I understand well enough to ask the question is, what are the team's feelings of default exports vs named exports?

File: client/src/Editor.jsx
Comment (gacetta) [inline client/src/Editor.jsx:1]: wow.  incredible job working through this.

**Review by gacetta — APPROVED** [review]
Nice job!

File: client/src/Editor.jsx (hunk 1 above)
Comment (clark-pang) [reply client/src/Editor.jsx:67]: I'm neutral, I left the boilerplate the way it was for this :)

File: client/src/App.jsx
[the remark this answers is not in the document]
Comment (clark-pang) [reply client/src/App.jsx:1]: Ah my bad, that must've been an artifact. I know for sure I'm using it in Editor.

**Pull request merged**